File: personalizatio-sdk/src/main/java/com/personalizatio/sdk/REES46.java
package com.personalizatio.sdk;

import android.content.Context;

/**
 * Created by Sergey Odintsov
 *
 * @author nixx.dj@gmail.com
 */
final public class REES46 extends SDK {

	public static final String TAG = "REES46";
	private static final String PREFERENCES_KEY = "rees46.sdk";
	public static final String NOTIFICATION_URL = "REES46_NOTIFICATION_URL";

	/**
	 * @param context application context
	 * @param shop_id Shop key
	 */
	protected REES46(Context context, String shop_id) {
		super(context, shop_id, Api.R46.class);
	}

	@Override
	protected String getPreferencesKey() {
		return PREFERENCES_KEY;
	}

	@Override
	public String getTag() {
		return TAG;
	}

	/**
	 * Initialize api
	 * @param shop_id Shop key
	 */
	public static void initialize(Context context, String shop_id) {
		instance = new REES46(context, shop_id);
	}

}

File: personalizatio-sdk/src/rees46/java/com/rees46/sdk/REES46.kt
package com.rees46.sdk

import android.content.Context
import com.personalizatio.BuildConfig
import com.personalizatio.SDK
import com.personalizatio.notification.NotificationHelper
import java.util.concurrent.Executors

class REES46 private constructor() : SDK() {

    companion object {
        const val TAG: String = "REES46"

        protected const val PREFERENCES_KEY: String = "rees46.sdk"
        protected val API_URL: String = when {
            BuildConfig.DEBUG -> "http://dev.api.rees46.com:8000/"
            else -> "https://api.rees46.ru/"
        }
        private val executorService = Executors.newFixedThreadPool(4)

        /**
         * Initialize api
         * @param context application context
         * @param shopId Shop key
         */
        fun initialize(context: Context, shopId: String, apiHost: String?) {
            if (!isInstanced()) {
                val apiUrl = apiHost?.let { "https://$it/" } ?: API_URL
                getInstance().initialize(context, shopId, apiUrl, TAG, PREFERENCES_KEY, "android")
            }

            // Дефолтное отображение сообщения без кастомизации
            getInstance().setOnMessageListener { data: Map<String, String> ->
                executorService.submit {
                    NotificationHelper.createNotification(
                        context = context,
                        data = data,
                        images = NotificationHelper.loadBitmaps(
                            urls = data[NotificationHelper.NOTIFICATION_IMAGES]
                        ),
                        currentIndex = 0
                    )
                }
            }
        }
    }
}
